File: VB/ReportService/ReportService.svc.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Data.Utils.ServiceModel
Imports DevExpress.XtraReports.Service
Imports CustomEditorSample
Imports DevExpress.XtraReports.UI

Namespace ReportService
	<SilverlightFaultBehavior> _
	Public Class ReportService
		Inherits DevExpress.XtraReports.Service.ReportService
		Shared Sub New()
			EditActionInvokerFactory.RegisterInvoker(Of ChangeImageSizeModeAction, ChangeImageSizeModeActionInvoker)()
		End Sub

		Protected Overrides Sub FillCustomProperties(ByVal reportName As String, ByVal control As XRControl, ByVal properties As System.Collections.Generic.IDictionary(Of String, Object))
			Dim pictureBox = TryCast(control, XRPictureBox)
			If pictureBox Is Nothing Then
				Return
			End If
			properties("Sizing") = pictureBox.Sizing
		End Sub
	End Class
End Namespace

File: CS/ReportService/ReportService.svc.cs
using DevExpress.Data.Utils.ServiceModel;
using DevExpress.XtraReports.Service;
using CustomEditorSample;
using DevExpress.XtraReports.UI;

namespace ReportService {
    [SilverlightFaultBehavior]
    public class ReportService : DevExpress.XtraReports.Service.ReportService {
        static ReportService() {
            EditActionInvokerFactory.RegisterInvoker<ChangeImageSizeModeAction, ChangeImageSizeModeActionInvoker>();
        }

        protected override void FillCustomProperties(string reportName, XRControl control, System.Collections.Generic.IDictionary<string, object> properties) {
            var pictureBox = control as XRPictureBox;
            if(pictureBox == null)
                return;
            properties["Sizing"] = pictureBox.Sizing;
        }
    }
}
